### ISSUE TITLE
Add support for filtering by file type

### ISSUE THREAD

**Issue by kauffj** [lbryio/lighthouse#111]
I believe this involves the following:

1) Reading the file's content type into the elastic search document. We currently store the file's media type (mime type) in the [content_type](https://github.com/lbryio/chainquery/blob/master/db/chainquery_schema.sql#L154) column on a claim.

2) Creating a mapping between mime/media types and more usable file types. While it's fine/good to support direct querying by media type, it is more useful to search by a generic type. So "video" might correspond to `video/mp4`, `video/x-msvideo`, etc.

Some suggested first types are: video, audio, written, CAD, archive

Presumably, these human readable types should _not_ be written to the document, instead the API can map/unmap these more usable terms in business logic.

3) Adding API support to perform queries by both the direct media type and the more generic types described in 2.

### Acceptance Criteria
1.
2.
3.

### Definition of Done
- [ ]  Tested against acceptance criteria
- [ ] Tested against the assumptions of the user story
- [ ] The project builds without errors
- [ ] Unit tests are written and passing
- [ ] Tests on devices/browsers listed in the issue have passed
- [ ] QA performed & issues resolved
- [ ] Refactoring completed
- [ ] Any configuration or build changes documented
- [ ] Documentation updated
- [ ] Peer Code Review performed

**Comment by alyssaoc** [comment]
@tiger5226 and @filipnyquist first we need a basic triage, is this issue workable/doable? What else do we need to know? Do you(either one or both) want to do it? There is a community member who may be interested, is this issue suitable for that situation?
We need to either:
1)  complete 111
2) Agree that 111 is workable and that lighthouse is in reasonable shape for a new person to attempt tackling this

Thoughts?

**Comment by tiger5226** [comment]
@alyssaoc We need to to know the precise mappings referenced by @kauffj. As long as how we do this is up to do us, this should be enough information to complete the issue. 

As for a new person tackling this, is this what we want? Why not have @filipnyquist or I work on it?

**Comment by alyssaoc** [comment]
@tiger5226 That is choice one, either one of you works on it. If you want to pass, you can, just confirm that it is workable. It opens up a lot of other development opportunities and is important but not urgent.

**Comment by kauffj** [comment]
I'm fine with anyone doing it. I believe @filipnyquist is expected to be unavailable for a few weeks as he is beginning university.

The background here is that I encouraged @jamesbiller to seek out a technical contributor who was familiar with maker (3d printing and more) use-cases. After some discussion, we realized that a lot of desired directions are blocked by this issue, like the ability to construct views that only show CAD content. Additionally, since I consider pursuing this domain to be in the experimental area, I'm hesitant to dedicate significant amounts of core contributor time to maker-specific issues. However, this feature is clearly of broad and general benefit, so I don't think that applies here.

So basically the options are:
1) Ensure this is well-spec'ed and use as an evaluation task
2) Beamer tackles this now (I'm fine with this if your slate isn't too full, I think several API features are already pending work at other levels)
3) Filip tackles once he is back to a more regular cadence

**Comment by kauffj** [comment]
Regarding creating the file type mappings, I think my initial list is a fine start. The engineer working on this should construct the specific mime type list.

**Comment by tiger5226** [comment]
@kauffj All clear! I will put this on my list. I run into situations sometimes where I am temporarily out of work for milestones and I pick from the queue. I like to have a secondary queue to fallback on. Right now that is api-forms ( did some more work there last weekend), chainquery, and lighthouse. So this is actually perfect for that fall back queue when I need to wait for responses or new issues.

**Comment by tiger5226** [comment]
merged and deployed with #144
